vanilla_bgv/encoding.py

```diff
@@ -0,0 +1,38 @@
+"""Routines for encoding and decoding messages as RLWE plaintexts for BGV.
+
+In BGV the plaintext space is the ring of polynomials
+
+    (Z/tZ)[x]/(x^N + 1)
+
+where N is a power of 2 and t > 2 is a prime.
```

Comment:
In BGV generally t can be a prime power (t^r) - it still has the same slot-wise encodings and rotations / automorphisms, but the calculation of the maps becomes a little more complicated

Reply:
I admit this is a simplification for this first version and for my own understanding, but I wonder which implementations are actually using the prime power version...

Reply:
HElib does!

Reply:
I went digging and confirmed OpenFHE does NOT. Hm